### DISCUSSION TITLE
I want to power off my display before deepsleep starts.

### DISCUSSION THREAD

**Discussion by Sfre12** [arendst/Tasmota#18619] [Q&A]
Hi
I am using a wemos D1 with Tasmota 12.x and script support.
additional I have connected a SSD1306 Display. This is working fine.
I am using deepsleep to save some energy. Now sometimes the dispay is still ON even if the Wemos is in Deepsleep. I don´t know how this happend, because nobody is switching on the display through web UI. So thats the first point...why the display is sometimes switching on in the night?
I already but the two sections with power 0 to the script, because I suggest sometimes the display is ON after deepsleep. (When I am looking to the device it never happends, mostly only over night)
>B
->power1 0
>BS
->power1 0

Is there any possibility to send the command "->power1 0" to swicth the display of within the script before the ESP is going to deepsleep?
I saw there is a trigger aviliable for Rules..."System#Save" Is something like this aviliable with script`s

**Comment by barbudor** [comment]
How long is your device awake ?
A standard usage of DeepSleep is
- Wake up
- Get sensor values, publish Teleperiod
- Go back to sleep
All this occurs generally in less than 14 seconds
So I don't think you need the display to be ON at any time.
Why now just systematically powering it OFF right at the beginning of the script ?

**Reply by Sfre12** [reply]
Hi barbudor.
Thanks for your replay!
--How long is your device awake ?
Maximum 30 seconds… look my script…
--A standard usage of DeepSleep is
yes. thats what I am doing.
--Wake up
--Get sensor values, publish Teleperiod (I am pulishing by script manualy, but think it is the same)
--Go back to sleep
I never want to use the display on. I installed an swtich with "deepsleep" to keep the device up if there is any need. (This is possible during daytime, because I need some other functions than later…)
So, but in general the device is going back to deepsleep, but as I wrote, the display keeps sometimes on. And not with correkt values…It looks like it is frozen with the “startup values”
Look my pic…
--All this occurs generally in less than 14 seconds
--So I don't think you need the display to be ON at any time.
--Why now just systematically powering it OFF right at the beginning of the script ?
I am powering of because I don`t know why the display is on sometimes.
even if I want to do something alse then switching display of...
The very important thing I want to know is how to get a "Trigger" functionality like with rules..
Do do any action before deepsleep!

best regards
Stephan

[Skript_108.txt](https://github.com/arendst/Tasmota/files/11435026/Skript_108.txt)
![DisplayOn](https://github.com/arendst/Tasmota/assets/79339733/8a2bd161-0542-46f7-bea3-e2d308001b97)
